Django Admin

CasinoRoyal666 opened the pull request:
feat: admin.py : Admin panel added and configurated. Also added serializers and Unit tests for them.
fix: model.py: Database field "total_price" now calculated based on OrderItems price and quantity

CasinoRoyal666 committed:
Add a basic admin panel with orders and events

CasinoRoyal666 committed:
feat: Add in events panel search, sort, ticket status. Add in every event field groups

CasinoRoyal666 committed:
feat: Add in orders panel search and filters. Also add a function for an order status color badge

CasinoRoyal666 committed:
fix: Fix a total_price field logic in model: now total price value tied to items and readonly in admin!

CasinoRoyal666 committed:
feat: Add serializers with Unit tests

valeryia-rusakova commented on events/tests.py:
```diff
@@ -1,3 +1,164 @@
+from decimal import Decimal
+from .models import Event, Order, OrderItem
+from .serializers import EventSerializer, OrderSerializer, OrderCreateSerializer
 
```
recommend splitting tests into tests/event/test_serializers.py and tests/order/ folders for better organization

valeryia-rusakova commented on events/tests.py:
```diff
@@ -1,3 +1,164 @@
+from .serializers import EventSerializer, OrderSerializer, OrderCreateSerializer
 
-# Create your tests here.
+
```
сonsider using pytest, pytest-django for writing tests, use @pytest.fixtures for creating test data and @pytest.mark.django_db

valeryia-rusakova commented on events/serializers.py:
```diff
@@ -0,0 +1,65 @@
+        model = Order
+        fields = ['customer_email', 'customer_name', 'customer_phone', 'items']
+    
+    def create(self, validated_data):
```
It’s better to keep business logic like creating orders outside the serializer. 
Serializers should handle validation and data formatting, put the creation logic in a separate service or function to keep code cleaner and easier to test.

CasinoRoyal666 committed:
fix: Fix unit_price field in models.py: now unit_price cost field based on Event.price field. unit_price changes apply when the Event changes

CasinoRoyal666 committed:
feat: Split tests into separate folders by model (events/orders).
Migrate to pytest framework
Add pytest fixtures and @pytest.mark.django_db for db tests
Add a service file with order creation logic

CasinoRoyal666 committed:
fix: Order create logic deleted from serializers. Order create business logic transfered into services

CasinoRoyal666 committed:
chore: update requirements file

CasinoRoyal666 committed:
chore: README now contains instructions for running tests

CasinoRoyal666 committed:
chore: Update readme, change pytest.ini location
